File: app/src/main/java/com/worldbuilder/mapgame/ui/dialogs/CustomizeWorldDialog.java
package com.worldbuilder.mapgame.ui.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatDialogFragment;

import com.worldbuilder.mapgame.databinding.DialogCustomizeWorldBinding;

public class CustomizeWorldDialog extends AppCompatDialogFragment {

    private CustomizeWorldDialogListener listener;
    private DialogCustomizeWorldBinding binding;

    public interface CustomizeWorldDialogListener {
        void onCreateWorld(float waterFrequency, float mountainFrequency);
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        try {
            listener = (CustomizeWorldDialogListener) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(context + " must implement CustomizeWorldDialogListener");
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(requireActivity());
        binding = DialogCustomizeWorldBinding.inflate(getLayoutInflater());

        builder.setView(binding.getRoot());

        binding.btnSave.setOnClickListener(v -> {
            float waterFrequency = binding.seekbarWaterFrequency.getProgress() / 100f;
            float mountainFrequency = binding.seekbarMountainFrequency.getProgress() / 100f;
            listener.onCreateWorld(waterFrequency, mountainFrequency);
            dismiss();
        });

        binding.btnCancel.setOnClickListener(v -> dismiss());

        return builder.create();
    }
}

File: app/src/main/java/com/worldbuilder/mapgame/ui/dialogs/CustomizeWorldDialog.kt
package com.worldbuilder.mapgame.ui.dialogs

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatDialogFragment
import com.worldbuilder.mapgame.databinding.DialogCustomizeWorldBinding
import java.lang.NullPointerException

interface CustomizeWorldDialogListener {
    fun onCreateWorld(waterFrequency: Float, mountainFrequency: Float)
}

class CustomizeWorldDialog : AppCompatDialogFragment() {
    private var listener: CustomizeWorldDialogListener? = null
    private lateinit var binding: DialogCustomizeWorldBinding

    fun setOnCreateWorldListener(listener: CustomizeWorldDialogListener) {
        this.listener = listener
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = DialogCustomizeWorldBinding.inflate(layoutInflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {

        binding.btnSave.setOnClickListener {
            val waterFrequency = binding.seekbarWaterFrequency.progress / 100f
            val mountainFrequency = binding.seekbarMountainFrequency.progress / 100f
            listener?.onCreateWorld(waterFrequency, mountainFrequency) ?: run {
                throw NullPointerException(
                    "required listener is null, make sure that you called " +
                            "CustomizeWorldDialog.setOnCreateWorldListener() method"
                )
            }
            dismiss()
        }
        binding.btnCancel.setOnClickListener { dismiss() }
    }
}